FS.common.js

```diff
@@ -180,6 +181,39 @@ var RNFS = {
     RNFSManager.stopDownload(jobId);
   },
 
+  uploadFiles(toUrl, files, options, begin, progress) {
+    var jobId = getJobId();
+    var subscriptionIos;
+
+    if (!begin) begin = (info) => {
+      console.log('Upload begun:', info);
+    };
+
+    if (begin) {
+      if (NativeAppEventEmitter.addListener)
```

Comment:
I think we should just check once beforehand whether `NativeAppEventEmitter.addListener` is available, instead of doing it every single time. It can be a constant value like:

`var NATIVE_ADD_LISTENER_AVAILABLE = !!NativeAppEventEmitter.addListener;`